src/lib.rs

```diff
@@ -348,6 +348,18 @@ pub trait Query<DB: Database>: Debug + Default + Sized + 'static {
 
     /// Internal struct storing the values for the query.
     type Storage: plumbing::QueryStorageOps<DB, Self> + Send + Sync;
+
+    /// Generated struct that contains storage for all queries in a group.
+    type GroupStorage;
+
+    /// Generated descriptor for all queries in a group.
+    type GroupDescriptor;
+
+    /// Extact storage for this query from the storage for its group.
+    fn storage(group_storage: &Self::GroupStorage) -> &Self::Storage;
+
+    /// Create group descriptor for this query.
+    fn descriptor(key: Self::Key) -> Self::GroupDescriptor;
```

Comment:
`fn group_descriptor`? I'd expect `Query::descriptor` to return `QureyDescriptor`

Reply:
Oh, the fact that `QueryDescriptor` is **larger** than `GroupDescriptor` is also confusing :)

Reply:
Heh, true. I suspect the names could use some work. OTOH, `QueryDescriptor` is a type end-users may actually interact with at some point, if only marginally.

Reply:
Maybe the names should be `GroupKey` and `DatabaseKey`

Reply:
these are all basically "upcasts" of key from a wider context -- a single query, to a group of queries, to the full database.

Reply:
I am fine with punting on this until we completely migrate to the new design, and *then* doing a one off run through all of the API to adjust trait names, visibility, etc.